src/Bicep.Core/Analyzers/Linter/LinterRuleBase.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    public abstract class LinterRuleBase : IBicepAnalyzerRule
+    {
+        public LinterRuleBase(string code, string ruleName, string description, string docUri,
+                          bool enableForEdit = true,
+                          bool enableForCLI = true,
+                          DiagnosticLevel diagnosticLevel = DiagnosticLevel.Warning)
+        {
+
+            this.ConfigHelper = new ConfigHelper();
+            this.AnalyzerName = LinterAnalyzer.AnalyzerName;
+            this.Code = code;
+            this.RuleName = ruleName;
+            this.Description = description;
+            this.DocumentationUri = docUri;
+
+            this.EnabledForCLI = enableForCLI;
+            this.EnabledForEditing = enableForEdit;
+            this.DiagnosticLevel = diagnosticLevel;
+
+            LoadConfiguration();
+        }
+
+        private readonly ConfigHelper ConfigHelper;
+        public string AnalyzerName { get; }
+        public string Code { get; }
+        public string RuleName { get; }
+
+        public const string RuleConfigSection = "Linter:Rules";
+
+        // TODO: Decide how we want to manage configuration
```

Comment:
Probably should discuss this now

Reply:
Discussion led us to expect that we will walk the folder tree from the location of the bicep file to find the first instance of bicepsettings.json and would merge those settings with the defaults. Defaults are now stored in embedded resource.